src/main/java/seedu/notor/logic/commands/PersonAddSubGroupCommand.java

```diff
@@ -47,31 +48,21 @@ public PersonAddSubGroupCommand(String personName, String groupName, String subG
         this.subGroupName = subGroupName;
     }
 
-
-
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        Person personToEdit = model.findPerson(personName);
-        if (personToEdit.getSubGroups().contains(groupName + "_" + subGroupName)) {
+        try {
+            Person personToEdit = model.findPerson(personName);
+            SuperGroup superGroup = model.findSuperGroup(groupName);
+            superGroup.addSubGroup(subGroupName);
+            SubGroup subGroup = superGroup.findSubGroup(subGroupName);
```

Comment:
Can be improved by having addSubGroup return the created subgroup, but I think leave for future update!